lib/core/connector_job.rb

```diff
@@ -67,7 +74,7 @@ def connector_snapshot
     end
 
     def connector_id
-      connector_snapshot[:id]
+      @elasticsearch_response[:_source][:connector][:id]
```

Comment:
Do you want to `dig` for nullability checks or it's okay to raise an exception if something's missing?

Reply:
It's okay to have a null-pointer issue here because I trust documents at this code level. If the document doesn't have `connector.id` then we can't process it and the job should fail.